src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -1359,7 +1371,15 @@ class spark_resource_adaptor final : public rmm::mr::device_memory_resource {
             throw std::invalid_argument(ss.str());
           }
           transition(thread->second, thread_state::THREAD_RUNNING);
+          // TODO why do we not set this to is_for_cpu
```

Comment:
I will have to look at the code again to try and remember. I think that this path is only taken for GPU allocations, but then why are we passing in `is_for_cpu`? I think it has something to do with `!was_recursive`, but like I said I am going to read through the code again to know for sure.

Reply:
I remember now. It is just be being a bit defensive and resetting the state when we transitioned to running. It should not need to be set at all, because it only matters when we are in a state that is in the middle of doing an allocation, and THREAD_RUNNING is not in the middle of doing any allocations.